### PR TITLE
don't pull images before docker-compose up, preventing image overwriting. 

### PR DESCRIPTION
**Pull request by changev**
https://rackhd.atlassian.net/browse/RAC-4847

docker-compose up is enough and 3rd images will be updated automatically

### CODE REVIEW

**Comment by panpan0000** [comment]
I knew it's so late for you to do this hot fix.
but to ensure it's easy to be reviewed for everybody.
let's create RAC and write detailed reason/test-report in PR

File: build-release-tools/post_test.sh [hunk 1]
```diff
@@ -381,18 +381,17 @@ post_test_docker() {
...
-            repo=${repo_tag%:*}
-            sed -i "s#rackhd/${repo}.*#rackhd/${repo_tag}#g" docker-compose-mini.yml
-        done
-        docker-compose -f docker-compose-mini.yml pull --ignore-pull-failures
-        docker-compose -f docker-compose-mini.yml up -d
-        mv docker-compose-mini.yml.bak docker-compose-mini.yml
-        waitForAPI
-        clean_all_containers
-    done < $buildRecord
+    image_list=$(head -n 1 $buildRecord)
```
Comment (PengTian0) [inline build-release-tools/post_test.sh]: Could you please add an example of buildRecord as a comment?

**Comment by PengTian0** [comment]
:+1: After add the comment of buildRecord

**Comment by panpan0000** [comment]
Let me explain:
original in ```while read -r LINE; do```
it will do same thing for 2.2.0 tag(line 1) and latest tag (line 2)
when doing the  ```docker-componse pull ``` for line 2(latest tag)
it will download the images w/ latest tag in the cloud(dockerhub) ,and it will override the local images with latest tag.
so the post-test are actually testing an old images on the cloud( but carried with latest tag).... instead of fresh-build locally.

and in the later docker-publish step, it use ```docker push``` to upload the local images to the cloud.
but the local "latest" are the same with cloud images...
so cloud images are never been updated 

👍  for this PR

**Comment by panpan0000** [comment]
@yyscamper  @johren  @anhou

File: build-release-tools/post_test.sh [hunk 2]
```diff
@@ -381,18 +381,22 @@ post_test_docker() {
...
+    # on-core:nightly on-tasks:nightly #
+    ####################################
+    image_list=$(head -n 1 $buildRecord)
+    cp docker-compose-mini.yml docker-compose-mini.yml.bak
+    for repo_tag in $image_list; do
+        repo=${repo_tag%:*}
+        sed -i "s#rackhd/${repo}.*#rackhd/${repo_tag}#g" docker-compose-mini.yml
+    done
+    docker-compose -f docker-compose-mini.yml pull --ignore-pull-failures
+    docker-compose -f docker-compose-mini.yml up -d
```
Comment (keedya) [inline build-release-tools/post_test.sh]: @changev, what is the reason of pulling here. If you are running the test on the same machine that built the docker images, u should be able to run docker up

Comment (panpan0000) [reply build-release-tools/post_test.sh]: @keedya  @changev ,sorry to jump in,  I guess to pull the other 3rd party images like mongo ?

Comment (keedya) [reply build-release-tools/post_test.sh]: @panpan0000 u don't need to run pull to get the third party images. It is pulled automatically when u run the build command. It is also pulled automatically when u run the up command and the images are missing

Comment (johren) [reply build-release-tools/post_test.sh]: I agree, it seems that the pull can cause issues with the pulling other images on top of the local images and the up should take care of the third party images.

**Comment by johren** [comment]
Something for consideration later, should we be pushing the images from the docker_build step to a staging repository (not DockerHub) or exporting them to a shared location to allow the build and post test stages to be done on different slaves and/or avoid a possible window of something overwriting the build images between these two steps?

File: build-release-tools/post_test.sh (hunk 2 above)
Comment (changev) [reply build-release-tools/post_test.sh]: @keedya 
Do you mean ```docker-compose build```?
We need to solve version generation and dependency, so didn't choose ```docker-compose build``` but ```docker build one by one```

**Comment by changev** [comment]
@johren 
Thanks!
I'm trying to enhance this PR and prevent any overwriting.

File: build-release-tools/post_test.sh (hunk 2 above)
Comment (keedya) [reply build-release-tools/post_test.sh]: @changev I am not sure what you mean, my comment is related to this line:
docker-compose -f docker-compose-mini.yml up -d
above should be enough to run the test

Comment (panpan0000) [reply build-release-tools/post_test.sh]: Sorry to jump in again.
@changev is doing test.
but I'm curious, considering mongo is never pulled in a new slave, should "docker-compose up" download mongo image?

Comment (changev) [reply build-release-tools/post_test.sh]: @keedya 
Yes, docker-compose -f docker-compose-mini.yml up is enough! Thanks

Comment (keedya) [reply build-release-tools/post_test.sh]: 👍

**Comment by johren** [comment]
I wouldn't complicate this one with any additional pushes or exports given the time-critical nature.  I just mentioned it as something to consider as a follow-on enhancement.

**Review by johren — APPROVED** [review]

File: build-release-tools/post_test.sh (hunk 2 above)
Comment (panpan0000) [reply build-release-tools/post_test.sh]: Aha, interesting. Yes. I just tried and @keedya  is right.

Then Why user still bother to use "pull" for either docker or docker-compose :-), they can directlly use "run" or "up" !

**Review by keedya — APPROVED** [review]

**Comment by changev** [comment]
test this please

**Comment by panpan0000** [comment]
@changev , can you please change the PR title to make it more accurate ?

**Review by anhou — APPROVED** [review]

**Pull request merged**